### ISSUE TITLE
Null arguments in the debugger launch.json result in a SystemOutOfRange Exception

### ISSUE THREAD

**Issue by nergmada** [microsoft/vscode-cpptools#7077]
Type: Debugger

**Describe the bug**
- OS and Version: Ubuntu Linux 20.10
- VS Code Version: 1.53.2
- C/C++ Extension Version: 1.2.2
- Other extensions you installed (and if the issue persists after disabling them): CMake 0.0.17, CMake Tools 1.6.0
- A clear and concise description of what the bug is.

I updated from v1.45 of VSCode to 1.53.2, since then, if I have blank arguments in my Launch.json I get SystemOutOfRange exceptions when trying to launch the debugger.

```
{
    "version": "0.2.0",
    "configurations": [
        {
            "name": "(gdb) Launch",
            "type": "cppdbg",
            "request": "launch",
            "program": "${command:cmake.launchTargetPath}",
            "args": [
                "${input:overlap}",
                "${input:domainName}", 
                "${input:problemName}"
            ],
            "stopAtEntry": false,
            "cwd": "${workspaceFolder}",
            "environment": [
                {
                    "name": "PATH",
                    "value": "$PATH:${command:cmake.launchTargetDirectory}"
            }],
            "externalConsole": false,
            "MIMode": "gdb",
            "miDebuggerPath": "/usr/bin/gdb",
            "setupCommands": [
                {
                    "description": "Enable pretty-printing for gdb",
                    "text": "-enable-pretty-printing",
                    "ignoreFailures": true
                }
            ]
        }
    ],
    "inputs": [{
        "type": "promptString",
        "id": "domainName",
        "description": "The domain to execute",
        "default": "./Testing/Adam/bin_packing/driverlogshift/domain.pddl"
    }, {
        "type": "promptString",
        "id": "problemName",
        "description": "The problem to execute",
        "default": "./Testing/Adam/bin_packing/driverlogshift/pfile1.pddl"
    }, {
        "type": "pickString",
        "id": "overlap",
        "description": "Disable no overlap?",
        "options": [
            "",
            "--disable-no-overlap-optimisation"
        ],
        "default": ""
    }]
}
```

In this example, I take a path to a binary provided by CMake Tools (see "program" key) and then pass in a number of arguments. Some of these arguments may be empty strings (e.g. if a flag hasn't been selected) I suspect what is happening is that null strings are being filtered out, but the length of the array of args is being maintained resulting in the launch crashing

**Comment by WardenGnaw** [comment]
Related: https://github.com/microsoft/MIEngine/pull/1080

This is fixed, but we have not released a newer version of the debug adapter recently.

**Comment by michelleangela** [comment]
Fix is released in the 1.3.0 release version https://github.com/microsoft/vscode-cpptools/releases/tag/1.3.0
